### PR TITLE
Widget Image Segmentation

### PR DESCRIPTION
**Pull request by mishig25**
### Implemented image segmentation widget.

Request to Inference API (same as input for Image Classification widget):
```js
const requestBody = { file }; // file is an image file
```

Response from Inference API:
```js
Array<{ 
    label: string; 
    score: number;
    mask: string 
}>
```
Sample output:
```js
[
    {"score": 0.9982, "label": "remote", "mask": "iVHIJKLMNOPQRSTUVWXYZabcd..."},
    {"score": 0.9955, "label": "couch", "mask": "iVHIJKLMNOPQRSTUVWXYZabcd..."},
    {"score": 0.9987, "label": "cat", "mask": "iVHIJKLMNOPQRSTUVWXYZabcd..."}
]
```

Screenshots:
 | success state                                                                                                                                                         | success state (mask selected)                                                                                                                                                            |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|Screenshot 2021-09-30 at 17 13 27  | Screenshot 2021-09-30 at 17 13 41 |

|  initial state   |  error state  | 
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------|
| Screenshot 2021-09-30 at 17 13 10 | Screenshot 2021-09-30 at 17 14 36 | 

Check out the [interactive netlify demo](https://616ec0f25dee3b009dc46ea5--huggingface-widgets.netlify.app/)

cc: @Narsil @osanseviero @NielsRogge 

Update 1: demo should be working on Safari as well

### CODE REVIEW

**Comment by julien-c** [comment]
This looks super cool!

(but the link to netlify is just regular object detection for now, no?)

**Comment by mishig25** [comment]
@julien-c try the netlify link now :)

**Comment by mishig25** [comment]
@julien-c try the demo now 👍  https://6156db84ceff2cbcaeac510c--huggingface-widgets.netlify.app/

**Comment by julien-c** [comment]
WOW 😮

**Comment by osanseviero** [comment]
Very excited about this!!! It looks too good

**Comment by gary149** [comment]
Awesome 😎. Found an issue on iOS: the browse image button works it opens my images library then I select an image but nothing happens.

**Comment by mishig25** [comment]
@gary149 indeed, in the [updated demo here](https://6156f83dc591ac04d90b6e4e--huggingface-widgets.netlify.app/), you'll get:
Screenshot 2021-10-01 at 13 58 49

**Comment by mishig25** [comment]
> Update 1: demo should be working on Safari as well

https://615a24ebbb69fc346548d87c--huggingface-widgets.netlify.app/

File: widgets/src/lib/InferenceWidget/shared/consts.ts
Comment (mishig25) [reply widgets/src/lib/InferenceWidget/shared/consts.ts]: Great points. I've applied suggestion b
> set the rgb value from a function that would be parsing the hexadecimal functions

I've removed the rgb from the const (now it only has {color, hex}). The reason 

I did **not** apply suggestion a
> get the hex value directly from the tailwind theme

because I didn't want to add tailwind dependency to moon-landing/server

File: widgets/src/lib/InferenceWidget/widgets/ImageSegmentationWidget/ImageSegmentationWidget.svelte
Comment (mishig25) [reply widgets/src/lib/InferenceWidget/widgets/ImageSegmentationWidget/ImageSegmentationWidget.svelte:192]: > we may want to put this in a shared utils function ?

I'll keep this polyfill in ImageSegmentationWidget.svelte since this polyfill fills only specific subset of createImageBitmap API that is only needed for this widget. 

> And don't forget to revoke the blobURIs, depending from where they come from you are possibly creating a memory leak.

I've updated the changes in the polyfill in[ this commit](https://github.com/huggingface/huggingface_hub/pull/378/commits/2f4ea5cd6227ae84d38e62248403ca8c5d4804c7). Removed the line: `dataURL = URL.createObjectURL(data);`, which was not used in ImgSegWidget and would have been a memory concern. Moreover, I've tested with Safari dev tools to make sure that garbage collection was correctly freeing memories being used in Polyfills

Comment (beurkinger) [reply widgets/src/lib/InferenceWidget/widgets/ImageSegmentationWidget/ImageSegmentationWidget.svelte:192]: Outstanding 🙏

**Comment by beurkinger** [comment]
I get an error message while trying to run your branch:
```
> huggingface-widgets@0.0.1 dev /Users/thibault/Documents/web/moon-landing/server/huggingface_hub/widgets
> svelte-kit dev

 > html:/Users/thibault/Documents/web/moon-landing/server/huggingface_hub/widgets/src/routes/index.svelte:4:13: error: Expected "from" but found "{"
    4 │ ... type { Model...
      ╵    
```

I think but we should put lang="ts" in every <script /> tag, as it's the recommend way of proceeding by the Svelte/SvelteKit documentation. Actually I think we had more or loess decided to do it with @Pierrci, but never got around to doing it ?

**Comment by mishig25** [comment]
@beurkinger please let me know if you have any other comments on this PR. 

In terms of:
> > huggingface-widgets@0.0.1 dev /Users/thibault/Documents/web/moon-landing/server/huggingface_hub/widgets
> svelte-kit dev

we should probably take care of it in different PR

**Review by beurkinger — APPROVED** [review]

**Comment by mishig25** [comment]
Will merge it after I merge https://github.com/huggingface/transformers/pull/14062

**Pull request merged**
